api/v1beta1/kafkacluster_types.go

```diff
@@ -67,7 +67,10 @@ type KafkaClusterSpec struct {
 	DisruptionBudget            DisruptionBudget        `json:"disruptionBudget,omitempty"`
 	RollingUpgradeConfig        RollingUpgradeConfig    `json:"rollingUpgradeConfig"`
 	// +kubebuilder:validation:Enum=envoy;istioingress
+	// If istioingress is selected than IstioControlPlane must be filled out.
 	IngressController string `json:"ingressController,omitempty"`
+	// IstioControlPlane is a reference to the istio controlplane deployment for envoy configuration. It must be specified if istio ingress is used.
```

Comment:
The istio control plane deployment or custom resource needs to be references. I think the later thus the comment should be explicit about it to avoid any confusion.